addons/payment_stripe_sca/models/payment.py

```diff
@@ -253,7 +253,7 @@ def _create_stripe_refund(self):
     def _stripe_form_get_tx_from_data(self, data):
         """ Given a data dict coming from stripe, verify it and find the related
         transaction record. """
-        reference = data.get('metadata', data).get("reference")
+        reference = data.get('metadata').get("reference") or data.get("reference")
```

Comment:
```suggestion
        reference = data.get('metadata', {}).get("reference") or data.get("reference")
```

For aviod `get("reference")`from `None`.

Or `metadata `always is into `data`?

Reply:
IMO, it is always in data otherwise it would not have failed in the first place...

Reply:
Better safe than sorry as anyway no one is acting quick here...who knows, maybe at some point Stripe is changing the API and it is not there anymore at some point 😉